File: examples/python/django/consistency.py
from django.db import transaction


def apply_subscription_update(event):
    with transaction.atomic():
        subscription = Subscription.objects.select_for_update().get(
            external_id=event.resource_id
        )

        if not subscription.can_transition_to(event.payload["status"]):
            return

        subscription.status = event.payload["status"]
        subscription.save(update_fields=["status"])

File: skills/webhook-consumer-architecture/examples/go/net-http/consistency.go
package examples

import "context"

func (s *SubscriptionService) ApplyUpdate(ctx context.Context, event NormalizedWebhookEvent) error {
	tx, err := s.DB.BeginTx(ctx, nil)
	if err != nil {
		return err
	}
	defer tx.Rollback()

	subscription, err := s.Repo.GetForUpdate(ctx, tx, event.ResourceID)
	if err != nil {
		return err
	}

	nextStatus := event.Payload["status"].(string)
	if !subscription.CanTransitionTo(nextStatus) {
		return tx.Commit()
	}

	if err := s.Repo.UpdateStatus(ctx, tx, subscription.ID, nextStatus); err != nil {
		return err
	}

	return tx.Commit()
}
